legend-depot-generations-api/src/main/java/org/finos/legend/depot/services/api/generations/FileGenerationsService.java

```diff
@@ -31,7 +30,12 @@ public interface FileGenerationsService
 
     Optional<DepotGeneration> getFileGenerationsByFilePath(String groupId, String artifactId, String versionsId, String filePath);
 
-    List<StoredFileGeneration> findByType(String groupId, String artifactId, String versionId, String type);
+    default List<StoredFileGeneration> findByType(String groupId, String artifactId, String versionId, String type)
+    {
+        return findByType(groupId, artifactId, versionId, type, null);
+    }
+
+    List<StoredFileGeneration> findByType(String groupId, String artifactId, String versionId, String type, String elementPath);
```

Comment:
can we rename this one to findByTypeAndElementPath , so it stays true to the function signature